plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -309,6 +334,18 @@ public void writeBlobAtomic(String blobName, InputStream inputStream, long blobS
         writeBlob(blobName, inputStream, blobSize, failIfAlreadyExists);
     }
 
+    @ExperimentalApi
+    @Override
+    public void writeBlobAtomicWithMetadata(
```

Comment:
Why do we need this method when `writeBlobWithMetadata` is also there?

Reply:
right, I'll remove this method as there is no implementation difference b/e `writeBlob` and `writeBlobAtomic`. Thanks!